### DISCUSSION TITLE
Understanding better how LOCAL_DOMAIN and WEB_DOMAIN works

### DISCUSSION THREAD

**Discussion by schiessle** [mastodon/mastodon#22696] [Q&A]
Hi all,

I setup my own Mastodon instance. Currently I'm in the final stage of setting everything up and preparing the switch. In order not to have to revert something when people started following me I would like to understand the way LOCAL_DOMAIN and WEB_DOMAIN works a little bit better.

My Mastodon server runs on mastodon.mydomain.org but I want to have UIDs like "user@mydomain.org". That's why I configured my server with this parameters:

````
LOCAL_DOMAIN=mydomain.org
WEB_DOMAIN=mastodon.mydomain.org
````

And creates the necessary redirect on mydomain.org. Everything works as expected. But I observed a few things:

1. While the web interface shows the right uid (user@mydomain.org) the Android client Fedilab shows user@mastodon.mydomain.org as user id at the profile on the left sidebar.
2. Also the official Android app for mastodon shows user@mastodon.mydomain.org on my profile page once connected and not user@mydomain.org
3. when I search for my new user from another Mastodon server, both seems to work (user@mydomain.org and user@mastodon.mydomain.org) which is not a surprise as it is just a redirect at the end.

But this raises some questions:

1. is the local_domain just something like a alias and the right address is always user@mastodon.mydomain.org?
2. What is stored on the remote server if someone follows me? Does it always default to the local domain or to the web domain? Or does it depend on what was entered when someone searched for my user?
4. Is this feature to separate between the local domain and the web domain a first class citizen in the mastodon world or rather a hack which server admins should avoid if possible.
5. Same as question 3 but with respect to the fediverse as a whole. How does software such a pixelfed peertube, friendica,... handle it. Do they all handle the distinction well or should I stick to one domain for web and local?

Thanks a lot!
Björn

**Comment by aubrielee** [comment]
I have this same question. Some other links I’ve found on the topic—not sure if you or others can make sense of them as applied to this question:
https://dallas.lu/en/use-your-primary-domain-as-an-mastodon-name
https://aeracode.org/2022/11/01/fediverse-custom-domains/
